main.rb

```diff
@@ -0,0 +1,140 @@
+require './book'
+require './person'
+require './rental'
+require './student'
+require './teacher'
+def add_student
+  puts 'Name'
+  name = gets.chomp
+  puts 'Age'
+  age = gets.chomp
+  puts 'Has parent permission (Y/N)'
+  has_permission = gets.chomp
+  has_permission = has_permission == 'y'
+  student = Student.new(age, '', name, has_permission)
+  puts 'Student created successfully'
+  student
+end
+
+def add_teacher
+  puts 'Name'
+  name = gets.chomp
+  puts 'Age'
+  age = gets.chomp
+  puts 'Specialization'
+  specialization = gets.chomp
+  teacher = Teacher.new(age, specialization, name, parent_permission: nil)
+  puts 'Teacher created successfully'
+  teacher
+end
```

Comment:
- [ ] This function produces an error of the wrong number of arguments, teacher add function should not have parent_permission parameter.